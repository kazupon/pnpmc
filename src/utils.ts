export function log(...args: unknown[]): void {
  console.log(...args)
}

export function fail(...messages: unknown[]): never {
  console.error(...messages)
  process.exit(1)
}
