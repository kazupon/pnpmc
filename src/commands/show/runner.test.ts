import { createCommandContext } from 'gunshi/context'
import path from 'node:path'
import { afterEach, expect, test, vi } from 'vitest'
import { defineMockLog } from '../../../test/utils.js'

afterEach(() => {
  vi.resetAllMocks()
})

test('default', async () => {
  const meta = (await import('./meta')).default
  const run = (await import('./runner')).default
  const utils = await import('../../utils')
  const log = defineMockLog(utils)
  // @ts-ignore
  const cwd = path.resolve(import.meta.dirname, '../../../test/fixtures/basic')
  const ctx = await createCommandContext({
    options: meta.options,
    values: {
      catalog: false,
      dependency: false
    },
    positionals: [],
    rest: [],
    args: [],
    tokens: [],
    command: { ...meta, run },
    omitted: true,
    commandOptions: { cwd, version: '0.0.0', name: 'pnpmc' }
  })

  await run(ctx)

  const message = log()
  expect(message).toContain('Defined catalogs')
  expect(message).toContain('Catalogable Dependencies')
  expect(message).toMatchSnapshot()
})

test('catalog only', async () => {
  const meta = (await import('./meta')).default
  const run = (await import('./runner')).default
  const utils = await import('../../utils')
  const log = defineMockLog(utils)
  // @ts-ignore
  const cwd = path.resolve(import.meta.dirname, '../../../test/fixtures/basic')
  const ctx = await createCommandContext({
    options: meta.options,
    values: {
      catalog: true,
      dependency: false
    },
    positionals: [],
    rest: [],
    args: [],
    tokens: [],
    commandOptions: { cwd, version: '0.0.0', name: 'pnpmc' },
    command: { ...meta, run },
    omitted: true
  })

  await run(ctx)

  const message = log()
  expect(message).toContain('Defined catalogs')
  expect(message).not.toContain('Catalogable Dependencies')
  expect(message).toMatchSnapshot()
})

test('dependency only', async () => {
  const meta = (await import('./meta')).default
  const run = (await import('./runner')).default
  const utils = await import('../../utils')
  const log = defineMockLog(utils)
  // @ts-ignore
  const cwd = path.resolve(import.meta.dirname, '../../../test/fixtures/basic')
  const ctx = await createCommandContext({
    options: meta.options,
    values: {
      catalog: false,
      dependency: true
    },
    positionals: [],
    rest: [],
    args: [],
    tokens: [],
    commandOptions: { cwd, version: '0.0.0', name: 'pnpmc' },
    command: { ...meta, run },
    omitted: true
  })

  await run(ctx)

  const message = log()
  expect(message).not.toContain('Defined catalogs')
  expect(message).toContain('Catalogable Dependencies')
  expect(message).toMatchSnapshot()
})

test('both option enable', async () => {
  const meta = (await import('./meta')).default
  const run = (await import('./runner')).default
  const utils = await import('../../utils')
  const log = defineMockLog(utils)
  // @ts-ignore
  const cwd = path.resolve(import.meta.dirname, '../../../test/fixtures/basic')
  const ctx = await createCommandContext({
    options: meta.options,
    values: {
      catalog: true,
      dependency: true
    },
    positionals: [],
    rest: [],
    args: [],
    tokens: [],
    commandOptions: { cwd, version: '0.0.0', name: 'pnpmc' },
    command: { ...meta, run },
    omitted: true
  })

  await run(ctx)

  const message = log()
  expect(message).toContain('Defined catalogs')
  expect(message).toContain('Catalogable Dependencies')
  expect(message).toMatchSnapshot()
})
