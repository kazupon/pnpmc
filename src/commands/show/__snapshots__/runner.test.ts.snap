// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`both option enable 1`] = `
"ğŸ“™ Defined catalogs in pnpm-workspace.yaml:
  (none)

ğŸ“¦ Catalogable Dependencies (1):
  typescript:
    /packages/package1 (package1) : ^5.7.3
    /packages/package2 (package2) : ^5.6.0

"
`;

exports[`catalog only 1`] = `
"ğŸ“™ Defined catalogs in pnpm-workspace.yaml:
  (none)
"
`;

exports[`default 1`] = `
"ğŸ“™ Defined catalogs in pnpm-workspace.yaml:
  (none)

ğŸ“¦ Catalogable Dependencies (1):
  typescript:
    /packages/package1 (package1) : ^5.7.3
    /packages/package2 (package2) : ^5.6.0

"
`;

exports[`dependency only 1`] = `
"ğŸ“¦ Catalogable Dependencies (1):
  typescript:
    /packages/package1 (package1) : ^5.7.3
    /packages/package2 (package2) : ^5.6.0

"
`;
